### PR TITLE
[Tooling] Use `buildkite_pipeline_upload` action to inline Buildkite release pipelines

### PR DESCRIPTION
**Pull request by iangmaia**
## Description

Android counterpart: https://github.com/wordpress-mobile/WordPress-Android/pull/21437
Related: pdccJN-xR-p2

This PR uses the `release-toolkit` action `buildkite_pipeline_upload` implemented on https://github.com/wordpress-mobile/release-toolkit/pull/597.

Our release pipelines (such as `complete-code-freeze`, `finalize-release`, etc) usually require to start a separate pipeline to make a release build.
The changes in this PR will result in add the steps of a release pipeline to the currently running build, without starting a new build altogether.

## Testing information
I've added (and later reverted) a test commit 7863f25931d5182ec67ebbb94a7aaff890b581e3 to test the `complete_code_freeze` lane.

This is a test build created using the Buildkite Web UI directly with `PIPELINE=release-pipelines/complete-code-freeze.yml`:
- https://buildkite.com/automattic/wordpress-ios/builds/24663

This PR can be tested in a similar way to the test above: creating a branch off this one, slightly modifying a release lane to prevent side effects and running the build directly on Buildkite.

### CODE REVIEW

File: fastlane/lanes/release.rb [hunk 1]
```diff
@@ -504,19 +504,29 @@
...
+      pipeline_file: BUILDKITE_RELEASE_PIPELINE,
+      environment: environment
+    )
+  else
+    build_url = buildkite_trigger_build(
+      buildkite_organization: BUILDKITE_ORGANIZATION,
+      buildkite_pipeline: BUILDKITE_PIPELINE,
+      branch: branch,
+      environment: environment,
+      pipeline_file: ,
```
Comment (dangermattic) [inline fastlane/lanes/release.rb:524]: <table data-meta="generated_by_pr-check">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/wordpress-mobile/WordPress-iOS/blob/2da8fb6267631eb8c8416b70907c34b9846f10a4/fastlane/lanes/release.rb#L524"></span>Style/HashSyntax: Include the hash value.</td>
    </tr>
  </tbody>
</table>

Comment (dangermattic) [inline fastlane/lanes/release.rb:524]: <table data-meta="generated_by_pr-check">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/wordpress-mobile/WordPress-iOS/blob/2da8fb6267631eb8c8416b70907c34b9846f10a4/fastlane/lanes/release.rb#L524"></span>Layout/SpaceBeforeComma: Space found before comma.</td>
    </tr>
  </tbody>
</table>

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-iOS/milestone/297">25.5</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by spencertransier — APPROVED** [review]

File: fastlane/lanes/release.rb [hunk 2]
```diff
@@ -504,19 +504,29 @@
...
-  build_url = buildkite_trigger_build(
-    buildkite_organization: BUILDKITE_ORGANIZATION,
-    buildkite_pipeline: BUILDKITE_PIPELINE,
-    branch: branch,
-    environment: { BETA_RELEASE: beta },
-    pipeline_file: 'release-builds.yml',
-    message: beta ? 'Beta Builds' : 'Release Builds'
-  )
+  environment = {
+    IS_BETA_RELEASE: beta,
```
Comment (mokagio) [inline fastlane/lanes/release.rb:508]: Just curious: Is the change from `BETA_RELEASE` to `IS_BETA_RELEASE` just polish you introduced or is there a wider discussion about it that I missed?

I don't mind either way, and I saw the only other occurrence in `release-build.yml` was updated 👌

**Review by mokagio — APPROVED** [review]
👍

File: fastlane/lanes/release.rb (hunk 2 above)
Comment (iangmaia) [reply fastlane/lanes/release.rb:508]: I think on WCiOS, this flag was named `BUILDKITE_BETA_RELEASE`, without really needing the `BUILDKITE_` prefix. So I renamed it to `IS_BETA_RELEASE` on that case. For consistency, I've also renamed this one to `IS_BETA_RELEASE`.

Comment (AliSoftware) [reply fastlane/lanes/release.rb:508]: FWIW if we're  gonna decide on a convention across our different repos on this, I'd vote for the env var to be named `IS_BETA_RELEASE` over just `BETA_RELEASE` to make it clear this is expected to be a boolean (and not e.g. the beta release version number as one might think when just reading the file)

**Pull request merged**